File: Leetcode 0448 Find All Numbers Disappeared in an Array [easy] [array]/Find All Numbers Disappeared in an Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        for (int i = 0; i < nums.length; i++) {
            int cur = Math.abs(nums[i]);
            nums[cur - 1] = -Math.abs(nums[cur - 1]);
        }
        
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                result.add(i + 1);
            }
        }
        
        return result;
    }
}

File: Leetcode 0448 Find All Numbers Disappeared in an Array [easy] [array]/Find All Numbers Disappeared in an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function(nums) {
    for (let i = 0; i < nums.length; i++) {
        let cur = Math.abs(nums[i]);
        nums[cur - 1] = -Math.abs(nums[cur - 1]);
    }
    
    let result = [];
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > 0) {
            result.push(i + 1);
        }
    }
    
    return result;
};